src/app/workouts/workoutCreate/WorkoutCreate.tsx

```diff
@@ -0,0 +1,142 @@
+
+  useEffect(() => {
+    // TODO: remove comment when proper data fetching exists
+    // const routineData = location.state?.routineData;
+    const routineData = mockRoutineData; // mocked routineData
+
+    if (routineData) {
+      const initialWorkout: Workout = {
+        name: routineData.routine.name,
+        date: new Date(),
+        comment: '',
+        routineId: routineData.routine.id,
+        exercises: routineData.exercises.map((routineExercise: RoutineExercise) => ({
+          exerciseId: routineExercise.exercise.id,
+          orderNumber: routineExercise.orderNumber,
+          notes: routineExercise.notes || '',
+          sets: Array.from(
+            { length: routineExercise.defaultsSets || 1 },
+            (_, index) => ({
+              setNumber: index + 1,
+              reps: routineExercise.defaultReps,
+              weight: routineExercise.defaultWeight,
+              restTime: routineExercise.defaultRestTime,
+              comment: '',
+            })
+          ),
+        })),
+      };
+      setWorkout(initialWorkout);
+    } else {
+      console.error('No routineData provided');
+    }
+  }, [location.state]);
+
+  const handleCreateWorkout = async () => {
+    if (workout) {
+      try {
+        const savedWorkout = await createWorkout(workout);
+        // TODO: remove comment when WorkoutDetails.tsx is created
+        // navigate(AppRoutes.WORKOUT_DETAILS.replace(':workoutId', savedWorkout.id!));
+      } catch (error) {
+        console.error('Error creating workout:', error);
+        alert('Failed to create workout.');
+      }
+    }
+  };
+
+  if (!workout) {
+    return <div>Loading...</div>;
+  }
+
+  return (
+    <div style={{ width: '100%' }}>
+      <div style={{ display: 'flex', alignItems: 'center', justifyContent: 'space-between', marginBottom: '16px' }}>
+        <Typography variant="h4" gutterBottom>
+          Create Workout from Routine
+        </Typography>
+        <Button
+          variant="contained"
+          color="primary"
+          onClick={handleCreateWorkout}
+        >
+          CREATE WORKOUT
+        </Button>
+      </div>
+
+      <Typography variant="h6">Workout Name:</Typography>
+      <Typography variant="body1">{workout.name}</Typography>
+
+      <Typography variant="h6" style={{ marginTop: '16px' }}>
+        Date:
+      </Typography>
+      <Typography variant="body1">{workout.date.toDateString()}</Typography>
+
+      {workout.comment && (
+        <>
+          <Typography variant="h6" style={{ marginTop: '16px' }}>
+            Comment:
+          </Typography>
+          <Typography variant="body1">{workout.comment}</Typography>
+        </>
+      )}
+
+      <Typography variant="h6" style={{ marginTop: '16px' }}>
+        Exercises:
+      </Typography>
+      <Grid container spacing={3} style={{ marginTop: '16px', width: '60%' }}>
+        {workout.exercises.map((exercise, index) => (
+          <Grid item xs={12} md={6} lg={4} key={index}>
```

Comment:
doing key as index is really bad approach.
It has to be unique indentifier to react that can distuinguish to rerender during reconciliation.
If you filter array react will still see same indexes to items that are different.
In those cases better to use unique indentifier (such as name, if you know it will be unique, or at best approach unique id)